Marlin/src/HAL/AVR/fast_pwm.cpp

```diff
@@ -247,13 +242,11 @@ void set_pwm_duty(const pin_t pin, const uint16_t v, const uint16_t v_size/*=255
     else {
       Timer timer = get_pwm_timer(pin);
       if (timer.n == 0) return; // Don't proceed if protected timer or not recognized
-      // Set compare output mode to CLEAR -> SET or SET -> CLEAR (if inverted)
-      #if ENABLED(DEBUG_PWM_INIT)
-        if (timer_freq[timer.n - 1] == 0) {       // If the timer is unconfigured and no freq is set then default PWM_FREQUENCY
-          set_pwm_frequency(pin, PWM_FREQUENCY);  // Set the frequency and save the value to the assigned index no.
-          SERIAL_ECHOLNPGM("PWM Timer ", timer.n, " Frequency Not Initialized!");
-        }
-      #endif
```

Comment:
As you may guess, this was set up as a debug option with the hope that we can eliminate this "on the fly" modification of the PWM frequency and get it set up robustly during initialization. This on-the-fly initialization is an evident admission that we are not in full control of the situation.

Reply:
We could setup a protected flag for timers and discard any frequency sets to them. That would address both the default and on the fly frequency issues.

Reply:
Or simpler yet we do the start up init which you suggested earlier. I  would lean towards that as the better solution.